contrib/images/e2e-initialization/init.Dockerfile

```diff
@@ -0,0 +1,39 @@
+FROM golang:1.21 as build-env
+
+ARG E2E_SCRIPT_NAME
+
+# Copy All
+WORKDIR /go/src/github.com/babylonlabs-io/babylon
+COPY ./ /go/src/github.com/babylonlabs-io/babylon/
```

Comment:
doesn't it need to be:

```
# Version to build. Default is empty
ARG VERSION

# Copy All
WORKDIR /go/src/github.com/babylonlabs-io/babylon
COPY ./ /go/src/github.com/babylonlabs-io/babylon/

# Handle if version is set
RUN if [ -n "${VERSION}" ]; then \
        git fetch origin ${VERSION}; \
        git checkout -f ${VERSION}; \
    fi

```

i.e we should be able to configure binary version here ?

Reply:
since we want this fixed version for testing, we don't need to have the version argument

Reply:
hmm correct me if I am wrong, but I have had imagined that this is needed similiary to `babylond-before-upgrade` step.

i.e for babylon before upgrade we call :
```
	docker build --tag babylonlabs-io/babylond-before-upgrade -f babylond/Dockerfile \
    --build-arg VERSION="${BABYLON_VERSION_BEFORE_UPGRADE}" ${BABYLON_FULL_PATH}
```
to actually build the image with some old version.

I thought this will happen here too i.e
1. First init chain with the `VERSION="${BABYLON_VERSION_BEFORE_UPGRADE}"`
2. then build image before upgrade also with `VERSION="${BABYLON_VERSION_BEFORE_UPGRADE}"` 
what am I missing ?

Reply:
We will have `VERSION` argument on main branch, that is to go to another version, here it will not be used to move to another version because this is already the fixed version we need to test 

But if you feel more comfortable I could add it

Reply:
lets add it for the sake of completeness, the worst case scenario it will be unused

Reply:
added version argument to dockerfile 
https://github.com/babylonlabs-io/babylon/pull/46/commits/26a3fc93d9cf559e695d76a676c519542dde35c5

At the makefile to build the docker I did not set any argument